FileSystem/FileIO.js

```diff
@@ -0,0 +1,263 @@
+  READ_WRITE: 'read_write'
+}
+
+/**
+  * File Input Output library
+  */
+var FileIO = {
+  api_open: api_open_,
+  api_close: api_close_,
+  api_available_file: api_available_file_,
+  api_lof: api_lof_,
+  api_eof: api_eof_,
+  api_print: api_print_,
+
+  open_files: {},
+
+  close_file: close_file_
+}
+
+/**
+  * Emulates VBA open statement API 
+  * 
+  * @todo Implement file locking
+  * @body File locking is not possible in Google Drive but it could be emulated by using PropertyServices.
+  * 
+  * @param {string} path The local file path
+  * @param {number} fileNumber File number
+  * @param {string} openMode Open Mode enumeration
+  * @param {string} accessMode Access Mode enumeration
+  * @param {string} lockMode lockMode enumeration
+  */
+function api_open_(path, fileNumber, openMode, accessMode, lockMode) {
+
+  // Set default arguments
+  openMode = openMode || OpenMode.RANDOM;
+  accessMode = accessMode || AccessMode.READ_WRITE;
+  lockMode = lockMode || LockMode.NONE;
+
+  // Check if file number is available
+  if (fileNumber in this.open_files)
+    throw Error("File Number: " + fileNumber + " is already being used");
+
+  // Check if file number is valid
+  if (fileNumber < 1 || fileNumber > 511)
+    throw Error("File Number: " + fileNumber + " is invalid. File numbers need to be an number value between 1 and 511");
+
+
+  var fileId;
+
+  // If file exists, get file id
+  // Else create and get file id
+  if (FileMapper.hasMapping(FileSystem.current_directory, path))
+    fileId = FileMapper.getFileId(FileSystem.current_directory, path);
+  else
+    fileId = create_file(FileSystem.current_directory, path, MimeType.PLAIN_TEXT);
+
+  // In memory file object
+  var file = {
+    fileId: fileId,
+    filePath: path,
+    accessMode: accessMode,
+    openMode: openMode,
+    lockMode: lockMode
+  };
+
+  // Set file content (In memory buffer)
+  switch (openMode) {
+    case OpenMode.INPUT:
+    case OpenMode.APPEND:
+    case OpenMode.RANDOM:
+      file.content = DriveApp.getFileById(file.fileId).getBlob().getDataAsString();
+      break;
+    case OpenMode.output:
+      file.content = "";
+      break;
+    case OpenMode.BINARY:
+      file.content = DriveApp.getFileById(file.fileId).getBlob().getBytes();
+  }
+
+  // Set file pointer
+  switch (openMode) {
+    case OpenMode.INPUT:
+    case OpenMode.RANDOM:
+    case OpenMode.OUTPUT:
+    case OpenMode.BINARY:
+      file.pointer = 0;
+      break;
+    case OpenMode.APPEND:
+      file.pointer = file.content.length;
+      break;
+  }
+
+  // Register Open File
+  this.open_files[fileNumber] = file;
+}
+
+/**
+  * Internal method to close file
+  * Saves the file content and closes the removed the file
+  * 
+  * @param {number} fileNumber File number to be closed
+  */
+function close_file_(fileNumber) {
+  if (!(fileNumber in this.open_files))
+    throw Error("File Number: " + fileNumber + " is not open");
+
+  var file = this.open_files[fileNumber];
+
```

Comment:
No need for empty lines between two statements unless there is a comment line

Reply:
Fixed